src/Marten/Linq/SqlGeneration/DocumentStatement.cs

```diff
@@ -33,22 +34,35 @@ protected override ISqlFragment buildWhereFragment(IMartenSession session)
             switch (WhereClauses.Count)
             {
                 case 0:
-                    where = Storage.DefaultWhereFragment();
-                    break;
+                    return Storage.DefaultWhereFragment();
 
                 case 1:
                     where = parser.Build(WhereClauses.Single());
+                    if (where is IgnoredWhereClauseFragment)
+                    {
+                        return Storage.DefaultWhereFragment();
+                    }
                     break;
 
                 default:
-                    var wheres = WhereClauses.Select(x => parser.Build(x)).ToArray();
-
-                    where = CompoundWhereFragment.And(wheres);
-                    break;
-
-
+                    var wheres = WhereClauses.Select(x => parser.Build(x))
+                        .Where(w => w is not IgnoredWhereClauseFragment)
+                        .ToArray();
+                    if (!wheres.Any())
+                    {
+                        return Storage.DefaultWhereFragment();
+                    }
+                    else if (wheres.Length == 1)
+                    {
+                        where = wheres[0];
```

Comment:
Shouldn't this be `where = parser.Build(wheres[0]);`?

Reply:
No. At this point, the where clause has already been parsed.

Reply:
Gotcha!